lib/shared-mongo-connections.js

```diff
@@ -0,0 +1,75 @@
+var mongoUrlParser = require('mongodb-uri');
+var async = require('async');
+var MongoClient = require('mongodb').MongoClient;
+var mongoose = require('mongoose');
+
+var MONGODB_DEFAULT_POOL_SIZE = 20;
+
+/**
+ * Get the url to connect to the admin mongodb specified in the given `config` object.
+ * @param {string} mongoConnectionString a mongodb url that should at least contain the mongodb hosts
+ * @param {object} mongoConf the mongodb configuration
+ * @param {string} mongoConf.admin_auth.user the admin username
+ * @param {string} mongoConf.admin_auth.pass the amdin user password
+ * @param {int} mongoConf.poolSize the poolSize of the mongodb connection. Default to 20.
+ * @return {string} the mongodb admin connection url
+ */
+function getAdminDbUrl(mongoConnectionString, mongoConf) {
```

Comment:
according to https://docs.mongodb.com/v2.4/reference/user-privileges/#any-database-roles, this types user have to be created in the admin database. Also because we are connecting to the db using a url, it's best just connect to the admin database right away. But the user here won't be a db admin user, it will be a user that only have "readWriteAnyDatabase" role.